src/dmd/root/longdouble.d

```diff
@@ -45,18 +46,23 @@ nothrow:
         else
             ld_set(&this, d);
     }
+    static if (real.sizeof > 8)
+        this(real r) { *cast(real*)&this = r; }
+
+    ushort exponent() const { return exp_sign & 0x7fff; }
+    bool sign() const { return (exp_sign & 0x8000) != 0; }
 
     void opAssign(float f) { ld_set(&this, f); }
     void opAssign(double f) { ld_set(&this, f); }
-    longdouble opNeg() const { return longdouble(mantissa, exp_sign ^ 0x8000); }
+    longdouble_soft opNeg() const { return longdouble_soft(mantissa, exp_sign ^ 0x8000); }
 
-    bool opEquals(const longdouble rhs) const { return this.ld_cmpe(rhs); }
-    int  opCmp(longdouble rhs) const { return this.ld_cmp(rhs); }
-    longdouble opAdd(longdouble rhs) const { return this.ld_add(rhs); }
-    longdouble opSub(longdouble rhs) const { return this.ld_sub(rhs); }
-    longdouble opMul(longdouble rhs) const { return this.ld_mul(rhs); }
-    longdouble opDiv(longdouble rhs) const { return this.ld_div(rhs); }
-    longdouble opMod(longdouble rhs) const { return this.ld_mod(rhs); }
+    bool opEquals(const longdouble_soft rhs) const { return this.ld_cmpe(rhs); }
```

Comment:
hmm, on this `rhs` is `const`, but not so for the other functions.

Reply:
Removed `const` for consistency as the argumets are passed by value anyway. I abuse the non-constness in some operations to avoid RVO ignoring the asm modifications.

Reply:
> RVO ignoring the asm modifications.

Please file a bug report about that.

Reply:
> Please file a bug report about that.

https://issues.dlang.org/show_bug.cgi?id=18758